lib/graphql/schema/default_type_error.rb

```diff
@@ -2,17 +2,14 @@
 module GraphQL
   class Schema
     module DefaultTypeError
-      def self.call(value, field, parent_type, query_ctx)
-        if field.type.kind.non_null?
-          if value.nil?
-            query_ctx.errors << GraphQL::InvalidNullError.new(parent_type.name, field.name, value)
-          else
-            # it was caused by GraphQL::ExecutionError
+      def self.call(type_error, ctx)
+        case type_error
+        when GraphQL::InvalidNullError
+          if !type_error.parent_error?
```

Comment:
I think this should be handled outside of the type_error callback, so the callback can only be called when there is an internal error to report.  Also, this check is only needed for the serial execution strategy, which is an inconsistency that shouldn't be exposed to the application code.